core/src/main/java/org/elasticsearch/discovery/zen/ping/unicast/UnicastZenPing.java

```diff
@@ -565,7 +565,8 @@ public void writeTo(StreamOutput out) throws IOException {
     }
 
     private PingResponse createPingResponse(DiscoveryNodes discoNodes) {
-        return new PingResponse(discoNodes.getLocalNode(), discoNodes.getMasterNode(), clusterName, contextProvider.nodeHasJoinedClusterOnce());
+        return new PingResponse(discoNodes.getLocalNode(), discoNodes.getMasterNode(), clusterName,
```

Comment:
This right here gets to the heart of one of my problems with this change. The shorter version is less verbose and more compatible with our enforced line-length limit (and so more likely to avoid odd wrapping like this). There is no official Java standard on prefixing the name with `get` and `set` (it comes from JavaBeans but we don't need to adhere to that in our codebase). I think the convention makes *some* sense in cases when there are mutable attributes on an object, but these methods are for properties on an immutable object. I note some examples from Java and the JDK: `String#length`, `List#size`, `Object#hashCode`. Sorry.